xmpp/xmpp.go

```diff
@@ -88,16 +92,22 @@ func (c *Conn) Auth(user, pass, resource string) {
 	fmt.Fprintf(c.outgoing, xmlIqSet, id(), NsIqAuth, user, pass, resource)
 }
 
+func (c *Conn) Oauth(token, resource string) {
+	msg := "\x00" + token + "\x00" + resource
+	b64 := base64.StdEncoding.EncodeToString([]byte(msg))
+	fmt.Fprintf(c.outgoing, xmlOauth, b64)
+}
+
 func (c *Conn) Features() *features {
 	var f features
 	c.incoming.DecodeElement(&f, nil)
 	return &f
 }
 
 func (c *Conn) Next() (xml.StartElement, error) {
+	for {
```

Comment:
You're totally right. `var element` doesn't need to sit outside of the `for` loop, it just needs to exist for the return value.